common/trinary/trit_tryte_sse42.h

```diff
@@ -0,0 +1,281 @@
+     *             ----------------        ------                  ------     ------     ------
+     *             ----------------        ------                  ------     ------     ------
+     * dataMid   = | f2 | f3 | g1 | ...... | k2 |  =>   midTrit  = | a2 | ... | f2 | ... | p2 |
+     *             ----------------        ------                  ------     ------     ------
+     *             ----------------        ------                  ------     ------     ------
+     * dataLast  = | k3 | l1 | l2 | ...... | p3 |       highTrit = | a3 | ... | f3 | ... | p3 |
+     *             ----------------        ------                  ------     ------     ------
+     */
+    __m128i lowTrit =
+        _mm_or_si128(_mm_shuffle_epi8(dataFirst, shuffleLow[0]),
+                     _mm_or_si128(_mm_shuffle_epi8(dataMid, shuffleLow[1]), _mm_shuffle_epi8(dataLast, shuffleLow[2])));
+    __m128i midTrit =
+        _mm_or_si128(_mm_shuffle_epi8(dataFirst, shuffleMid[0]),
+                     _mm_or_si128(_mm_shuffle_epi8(dataMid, shuffleMid[1]), _mm_shuffle_epi8(dataLast, shuffleMid[2])));
+    __m128i highTrit = _mm_or_si128(
+        _mm_shuffle_epi8(dataFirst, shuffleHigh[0]),
+        _mm_or_si128(_mm_shuffle_epi8(dataMid, shuffleHigh[1]), _mm_shuffle_epi8(dataLast, shuffleHigh[2])));
+    /* lowResult = (lowTrit) */
+    __m128i lowResult = lowTrit;
+    /* midResult = (midTrit * 3) */
+    __m128i midResult = _mm_or_si128(
+        _mm_and_si128(byteInterMask,
+                      _mm_mullo_epi16(_mm_and_si128(midTrit, byteInterMask),
+                                      _mm_set_epi16(0x0003, 0x0003, 0x0003, 0x0003, 0x0003, 0x0003, 0x0003, 0x0003))),
+        _mm_andnot_si128(byteInterMask, _mm_mullo_epi16(_mm_and_si128(midTrit, ~byteInterMask),
+                                                        _mm_set_epi16(0x0003, 0x0003, 0x0003, 0x0003, 0x0003, 0x0003,
+                                                                      0x0003, 0x0003))));
+    /* highResult = (highTrit * 9) */
+    __m128i highResult = _mm_or_si128(
+        _mm_and_si128(byteInterMask,
+                      _mm_mullo_epi16(_mm_and_si128(highTrit, byteInterMask),
+                                      _mm_set_epi16(0x0009, 0x0009, 0x0009, 0x0009, 0x0009, 0x0009, 0x0009, 0x0009))),
+        _mm_andnot_si128(byteInterMask, _mm_mullo_epi16(_mm_and_si128(highTrit, ~byteInterMask),
+                                                        _mm_set_epi16(0x0009, 0x0009, 0x0009, 0x0009, 0x0009, 0x0009,
+                                                                      0x0009, 0x0009))));
+    /* alphabetOffset = (lowResult + midResult + highResult) */
+    __m128i alphabetOffset = _mm_add_epi8(lowResult, _mm_add_epi8(midResult, highResult));
+    /* Check whether the offset is < 0 */
+    __m128i tmpMask = _mm_cmplt_epi8(alphabetOffset, _mm_set_epi32(0, 0, 0, 0));
```

Comment:
You can get rid of negative values (and hence no need to do this check) by adding 1 to the input `data`. `tryteAlphabet` would, of course, change.

Reply:
I add 13 to the final offset result to eliminate negative value and modify the `tryteAlphabet` as well.